### PR TITLE
Implement Monorepo and add Framework and Logging

### PR DESCRIPTION
**Pull request by ZHollingshead**
This merges in the work to implement a mono-repo for Aurelia.

This monorepo will cover the `@aurelia` organization on NPM: (i.e. `@aurelia/framework`, `@aurelia/logging`, `@aurelia/router`, etc).

## What has been done
- Implemented a monorepo
- Added framework as a package
- Added logging as a package
- Converted from Javascript to Typescript

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,191 +1,21 @@
 {
-  "name": "aurelia-framework",
-  "version": "1.1.5",
-  "description": "The aurelia framework brings together all the required core aurelia libraries into a ready-to-go application-building platform.",
-  "keywords": [
-    "aurelia",
-    "framework",
-    "spa"
-  ],
+	"name": "aurelia-ux",
```
Comment (EisenbergEffect) [inline package.json]: Seems like the name, bugs, etc. aren't quite right

File: packages/framework/package.json [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    "build": "concurrently \"npm run build:amd\" \"npm run build:commonjs\" \"npm run build:es2015\" \"npm run build:es2017\" \"npm run build:native-modules\" \"npm run build:system\"",
+    "predoc": "cross-env rimraf doc/api.json && rimraf dist/doc-temp && tsc --project tsconfig.json --outFile dist/doc-temp/aurelia-framework.js && node doc/shape-defs && copyfiles tsconfig.json dist/doc-temp",
+    "doc": "cross-env typedoc --json doc/api.json --excludeExternals --includeDeclarations --mode modules --target ES6 --name aurelia-framework-docs dist/doc-temp/",
+    "postdoc": "cross-env node doc/shape-doc && rimraf dist/doc-temp",
+    "changelog": "cross-env conventional-changelog -p angular -i doc/CHANGELOG.md -s",
+    "bump-version": "npm --no-git-tag-version version",
+    "preprepare-release": "cross-env npm run test",
+    "prepare-release": "cross-env npm run changelog && npm run build && npm run doc"
+},
+  "jspm": {
```
Comment (EisenbergEffect) [inline packages/framework/package.json]: @PWKad Do we need the jspm configuration anymore? Can it all be determined through the normal package json configuration? For vNext, I'd like to be able to remove this if possible.

File: packages/framework/package.json [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    "karma-jasmine": "^1.1.1",
+    "karma-requirejs": "^1.1.0",
+    "requirejs": "^2.3.5",
+    "requirejs-text": "^2.0.15",
+    "rimraf": "^2.6.2",
+    "tslint": "^5.8.0",
+    "typedoc": "^0.9.0",
+    "typescript": "^2.6.2"
+  },
+  "aurelia": {
```
Comment (EisenbergEffect) [inline packages/framework/package.json]: Going forward, we can remove the entire aurelia section of the package.json.

File: packages/framework/test/framework-configuration.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
 import './setup';
-import {FrameworkConfiguration} from '../src/framework-configuration';
-import {Aurelia} from '../src/aurelia';
-import {Metadata} from 'aurelia-metadata';
+import { FrameworkConfiguration } from '../src/framework-configuration';
+import { Aurelia } from '../src/aurelia';
+// import { Metadata } from 'aurelia-metadata';
```
Comment (EisenbergEffect) [inline packages/framework/test/framework-configuration.ts]: What's happening with the commented out code?

File: packages/logging/src/log-level.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Specifies the available logging levels.
+ */
+export interface LogLevel {
```
Comment (EisenbergEffect) [inline packages/logging/src/log-level.ts]: For TypeScript, should this be converted into an Enum?

File: package.json (hunk 1 above)
Comment (ZHollingshead) [reply package.json]: Copy paste got me.

File: packages/logging/src/log-level.ts (hunk 5 above)
Comment (ZHollingshead) [reply packages/logging/src/log-level.ts]: I think this affects custom logging and is best left as an interface. To my knowledge you add a new level to an Enum in typescript by declaring a second Enum of the same name. I think that would be difficult to do dynamically.

`LogLevel[name] = value;` for example would not work, which is how the current `addCustomLevel` function operates. Though perhaps there is an easy way of adding additional values to an Enum that I am not aware of.

File: packages/framework/package.json (hunk 2 above)
Comment (ZHollingshead) [reply packages/framework/package.json]: If it is determined this can be removed, should we do this on the UX repo as well?

Comment (EisenbergEffect) [reply packages/framework/package.json]: Yes.
@PWKad Any comments on this?

Comment (ZHollingshead) [reply packages/framework/package.json]: @EisenbergEffect @PWKad Any chance to look at this?

**Comment by cluka** [comment]
Nice! Especially the conversion to typescript

**Comment by bigopon** [comment]
If we keep distributing same JS at the same npm packages, won't this be breaking change ?

**Comment by EisenbergEffect** [comment]
Closing this for now since we're going to focus on vNext and do the full TS and monorepo implementation there.

**Pull request closed without merging**